ckan/controllers/feed.py

```diff
@@ -145,7 +145,7 @@ def _create_atom_id(resource_path, authority_name=None, date_string=None):
         # This is best we can do, and if the site_url is not set, then
         # this still results in an invalid feed.
         site_url = config.get('ckan.site_url', '')
-        return '/'.join([site_url, resource_path])
+        return urlparse.urljoin(site_url, resource_path)
```

Comment:
That sounds right. IIUC The intent of this code is to generate a url back to a feed controller method, so we should be able to use the parameters passed to that controller method and the name of the controller to call url_for to generate the correct url